lib/segment/src/index/field_index/map_index/mmap_map_index.rs

```diff
@@ -0,0 +1,226 @@
+const HASHMAP_PATH: &str = "values_to_points.bin";
+const CONFIG_PATH: &str = "mmap_field_index_config.json";
+
+pub struct MmapMapIndex<N: MapIndexKey + Key + ?Sized> {
+    path: PathBuf,
+    value_to_points: MmapHashMap<N>,
+    point_to_values: MmapPointToValues<N>,
+    deleted: MmapBitSliceBufferedUpdateWrapper,
+    deleted_count: usize,
+    total_key_value_pairs: usize,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct MmapMapIndexConfig {
+    total_key_value_pairs: usize,
+}
+
+impl<N: MapIndexKey + Key + ?Sized> MmapMapIndex<N> {
+    pub fn load(path: &Path) -> OperationResult<Self> {
+        let hashmap_path = path.join(HASHMAP_PATH);
+        let deleted_path = path.join(DELETED_PATH);
+        let config_path = path.join(CONFIG_PATH);
+
+        let config: MmapMapIndexConfig = read_json(&config_path)?;
+
+        let hashmap = MmapHashMap::open(&hashmap_path)?;
+        let point_to_values = MmapPointToValues::open(path)?;
+
+        let deleted = mmap_ops::open_write_mmap(&deleted_path)?;
+        let deleted = MmapBitSlice::from(deleted, 0);
+        let deleted_count = deleted.count_ones();
+
+        Ok(Self {
+            path: path.to_path_buf(),
+            value_to_points: hashmap,
+            point_to_values,
+            deleted: MmapBitSliceBufferedUpdateWrapper::new(deleted),
+            deleted_count,
+            total_key_value_pairs: config.total_key_value_pairs,
+        })
+    }
+
+    pub fn build(
+        path: &Path,
+        point_to_values: Vec<Vec<N::Owned>>,
+        values_to_points: HashMap<N::Owned, Vec<PointOffsetType>>,
+    ) -> OperationResult<Self> {
+        let hashmap_path = path.join(HASHMAP_PATH);
+        let deleted_path = path.join(DELETED_PATH);
+        let config_path = path.join(CONFIG_PATH);
+
+        atomic_save_json(
+            &config_path,
+            &MmapMapIndexConfig {
+                total_key_value_pairs: point_to_values.iter().map(|v| v.len()).sum(),
+            },
+        )?;
+
+        MmapHashMap::create(
+            &hashmap_path,
+            values_to_points
+                .iter()
+                .map(|(value, ids)| (<N as MapIndexKey>::from_owned(value), ids.iter().copied())),
+        )?;
+
+        MmapPointToValues::<N>::from_iter(
+            path,
+            point_to_values.iter().enumerate().map(|(idx, values)| {
+                (
+                    idx as PointOffsetType,
+                    values
+                        .iter()
+                        .map(|value| N::into_referenced(N::from_owned(value))),
+                )
+            }),
+        )?;
+
+        {
+            const BITS_IN_BYTE: usize = 8;
+            let deleted_flags_count = point_to_values.len();
+            let deleted_file = create_and_ensure_length(
+                &deleted_path,
+                BITS_IN_BYTE
+                    * BITS_IN_BYTE
+                    * deleted_flags_count.div_ceil(BITS_IN_BYTE * BITS_IN_BYTE),
```

Comment:
Not sure what happen here. Maybe you want one of the following?

```rust
deleted_flags_count.div_ceil(u8::BITS)
// or
deleted_flags_count.div_ceil(u8::BITS).next_multiple_of(size_of::<usize>())
```

Reply:
done